dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4380,6 +4380,419 @@ class FunctionRepeat : public IFunction
+                auto substr = substr_const_string->getValue<String>();
+                constConst(str, substr, val_num, res_vec);
+            }
+            else
+            {
+                const auto * substr_vector_string = checkAndGetColumn<ColumnString>(c1_col_column.get());
+                const auto * str_const_string_value = checkAndGetColumn<ColumnString>(str_const_string->getDataColumnPtr().get());
+                constVector(str_const_string_value->getChars(), str_const_string_value->getOffsets(), substr_vector_string->getChars(), substr_vector_string->getOffsets(), val_num, res_vec);
+            }
+        }
+        else
+        {
+            const auto * str_vector_string = checkAndGetColumn<ColumnString>(c0_col_column.get());
+            if (const auto * substr_const_string = checkAndGetColumnConst<ColumnString>(c1_col_column.get()))
+            {
+                const auto * substr_const_string_value = checkAndGetColumn<ColumnString>(substr_const_string->getDataColumnPtr().get());
+                vectorConst(str_vector_string->getChars(), str_vector_string->getOffsets(), substr_const_string_value->getChars(), substr_const_string_value->getOffsets(), val_num, res_vec);
+            }
+            else
+            {
+                const auto * substr_vector_string = checkAndGetColumn<ColumnString>(c1_col_column.get());
+                vectorVector(str_vector_string->getChars(), str_vector_string->getOffsets(), substr_vector_string->getChars(), substr_vector_string->getOffsets(), val_num, res_vec);
+            }
+        }
+
+        block.getByPosition(result).column = std::move(col_res);
+    }
+
+private:
+    static Int64 getPositionUTF8(const String & c1_str, Int64 idx)
+    {
+        if (idx == -1)
+            return 0;
+
+        const auto * data = reinterpret_cast<const UInt8 *>(c1_str.data());
+        return static_cast<size_t>(UTF8::countCodePoints(data, idx) + 1);
+    }
+
+    static Int64 find(
+        const ColumnString::Chars_t & col_vector_str_value,
+        const ColumnString::Offset & str_start_offset,
+        const ColumnString::Offset & str_end_offset,
+        const ColumnString::Chars_t & col_vector_substr_value,
+        const ColumnString::Offset & substr_start_offset,
+        const ColumnString::Offset & substr_end_offset)
+    {
+        unsigned int sub_search_index = 0;
+        unsigned int find_in_size = 0;
+        Int64 res = 0;
+        ColumnString::Offset substr_size = substr_end_offset - substr_start_offset - 1;
+        if (substr_size == 0)
+        {
+            return 1;
+        }
+
+        for (ColumnString::Offset i = str_start_offset; i < str_end_offset; ++i)
+        {
+            auto char_str = col_vector_str_value[i];
+            auto char_substr = col_vector_substr_value[substr_start_offset + sub_search_index];
+            if (char_str <= 'Z' && char_str >= 'A')
+            {
+                char_str = char_str - ('Z' - 'z');
```

Comment:
Write Z - z as hard code.  such as `ColumnString::Chars_t diff = 'Z' - 'z'`, and we can write this line as `char_str = char_str - diff;`